readers/mutation_reader.cc

```diff
@@ -236,18 +275,35 @@ bool mutation_fragment_stream_validating_filter::operator()(mutation_fragment_v2
 }
 
 bool mutation_fragment_stream_validating_filter::operator()(mutation_fragment::kind kind, position_in_partition_view pos) {
-    return (*this)(to_mutation_fragment_kind_v2(kind), pos);
+    return (*this)(to_mutation_fragment_kind_v2(kind), pos, {});
 }
 
 bool mutation_fragment_stream_validating_filter::operator()(const mutation_fragment_v2& mv) {
-    auto valid = (*this)(mv.mutation_fragment_kind(), mv.position());
-    if (valid && mv.is_range_tombstone_change()) {
-        _current_tombstone = mv.as_range_tombstone_change().tombstone();
-    }
-    return valid;
+    return (*this)(mv.mutation_fragment_kind(), mv.position(),
+            mv.is_range_tombstone_change() ? std::optional(mv.as_range_tombstone_change().tombstone()) : std::nullopt);
 }
```

Comment:
How about defining the this as a mutation_fragment_v2 method?

Reply:
I don't think it makes sense to do so to save an if on a single user. `mutation_fragment_v2` should have getters only for things that are common among all mutation fragment kinds, like `position()`.

Reply:
FWIW, there's another user in `mutation_fragment_stream_validator::reset`.
I would have suggested to define the getter that returns an optional range_tombstone their if this was the same class.